data_structures/binary_search_tree.cpp

```diff
@@ -122,9 +159,72 @@ void Post(node *n) {
     }
 }
 
+/** test traverse a tree with Post order
+ * and save the traverse node to an array.
+ * \param[in] n the node pointer of a tree to start traverse
+ * \param[out] arr the output array that save the traversed tree nodes pointers
+ */
+void testInOrderTraverse(node *n, std::vector<int> &arr) {
```

Comment:
```
void testInOrderTraverse(node *n, std::vector<int> &arr) {
```
This line is the error. the error describes that `arr` must be a const reference. If it is not, use a pointer.
```suggestion
void testInOrderTraverse(node *n, std::vector<int> *arr) {
```